File: c#/src/Docmost.Sdk/Services/WorkspaceApi.cs
using Docmost.Sdk.Http;
using Docmost.Sdk.Models;
using System.Net.Http.Headers;
using System.Net.Http.Json;
using System.Text.Json;

namespace Docmost.Sdk.Services;

public sealed class WorkspaceApi(DocmostHttpClient http)
{
    private readonly DocmostHttpClient _http = http;

    public async Task<ApiResponse<JsonElement?>> GetWorkspacePublicInfoAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/public", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetWorkspaceInfoAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/info", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateWorkspaceAsync(UpdateWorkspaceDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/update", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetWorkspaceMembersAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/members", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeactivateWorkspaceMemberAsync(CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/members/deactivate", null, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> DeleteWorkspaceMemberAsync(RemoveWorkspaceUserDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/members/delete", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> UpdateWorkspaceMemberRoleAsync(UpdateWorkspaceUserRoleDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/members/change-role", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetInvitationsAsync(PaginationOptions request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetInvitationByIdAsync(InvitationIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/info", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> InviteUserAsync(InviteUserDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/create", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> ResendInviteAsync(InvitationIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/resend", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> RevokeInviteAsync(InvitationIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/revoke", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> AcceptInviteAsync(AcceptInviteDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/accept", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> GetInviteLinkAsync(InvitationIdDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/invites/link", request, cancellationToken).ConfigureAwait(false);
    }

    public async Task<ApiResponse<JsonElement?>> CheckHostnameAsync(CheckHostnameDto request, CancellationToken cancellationToken = default)
    {
        return await _http.PostAsync<JsonElement?>("workspace/check-hostname", request, cancellationToken).ConfigureAwait(false);
    }

}

File: typescript/src/services/workspace-api.ts
/** Generated API client. */
import type { ApiResponse } from '../types.js';
import type { DocmostHttpClient } from '../http-client.js';
import type { AcceptInviteDto, CheckHostnameDto, InvitationIdDto, InviteUserDto, PaginationOptions, RemoveWorkspaceUserDto, UpdateWorkspaceDto, UpdateWorkspaceUserRoleDto } from '../models/index.js';


export class WorkspaceApi {
  constructor(private readonly http: DocmostHttpClient) {}

  async getWorkspacePublicInfo(): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/public`, undefined);
  }

  async getWorkspaceInfo(): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/info`, undefined);
  }

  async updateWorkspace(request: UpdateWorkspaceDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/update`, request);
  }

  async getWorkspaceMembers(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/members`, request);
  }

  async deactivateWorkspaceMember(): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/members/deactivate`, undefined);
  }

  async deleteWorkspaceMember(request: RemoveWorkspaceUserDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/members/delete`, request);
  }

  async updateWorkspaceMemberRole(request: UpdateWorkspaceUserRoleDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/members/change-role`, request);
  }

  async getInvitations(request: PaginationOptions): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites`, request);
  }

  async getInvitationById(request: InvitationIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/info`, request);
  }

  async inviteUser(request: InviteUserDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/create`, request);
  }

  async resendInvite(request: InvitationIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/resend`, request);
  }

  async revokeInvite(request: InvitationIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/revoke`, request);
  }

  async acceptInvite(request: AcceptInviteDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/accept`, request);
  }

  async getInviteLink(request: InvitationIdDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/invites/link`, request);
  }

  async checkHostname(request: CheckHostnameDto): Promise<ApiResponse<unknown>> {
    return this.http.post(`workspace/check-hostname`, request);
  }

}
